api/fossa/upload.go

```diff
@@ -55,6 +55,12 @@ func Upload(fetcher, project, revision, title, branch string, data []SourceUnit)
 	if branch != "" {
 		q.Add("branch", branch)
 	}
+	if projectURL != "" {
```

Comment:
I think we should refactor optional properties (title, branch, projectURL, jiraProjectKey) into a separate `type UploadOptions struct`. Thoughts?